### ISSUE TITLE
debian package doesn't build on wheezy / jessie (7/8)

### ISSUE THREAD

**Issue by tpdownes** [github/git-lfs#1050]
On Debian 7 platform, I have installed git, dh-golang and golang-go from the wheezy-backports repository. On Debian 8, these are installed straight from jessie.

It would be easier for me if you you deployed the software with a properly signed git-lfs repository, even if only for one package. It is super easy with aptly. Downloading .debs is not a very normal large-scale way of doing business even if it's normal for Joe Sixpack. Without a repo, my choices are to resign your package (with aptly or something) or to re-build it. The partners I work with who maintain our internal repos have the infrastructure to build packages and integrate them into existing repos so therefore we choose to re-build. But the packages have to actually build.

It looks to me like the problem is in the post-build testing (TestExistingUpload). Output of ```debuild -us -uc``` is below. By-eye the output was identical on Debian 8 and 7. I show the output of Debian 8. Results were identical (by-eye) on the 1.1.1 and the 1.1.1 pre-push-tracing tar balls.

```
=== RUN TestExistingUpload
--- FAIL: TestExistingUpload (0.00 seconds)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x5ff147]

goroutine 174 [running]:
runtime.panic(0x7ea460, 0xc5b2d3)
	/usr/lib/go/src/pkg/runtime/panic.c:279 +0xf5
testing.func·006()
	/usr/lib/go/src/pkg/testing/testing.go:416 +0x176
runtime.panic(0x7ea460, 0xc5b2d3)
	/usr/lib/go/src/pkg/runtime/panic.c:248 +0x18d
github.com/github/git-lfs/localstorage.localObjectDir(0x0, 0x8afd30, 0x28, 0x0, 0x0)
	/root/build/git-lfs-1.1.1-pre-push-tracing/obj-x86_64-linux-gnu/src/github.com/github/git-lfs/localstorage/localstorage.go:57 +0x127
github.com/github/git-lfs/localstorage.(*LocalStorage).BuildObjectPath(0x0, 0x8afd30, 0x28, 0x0, 0x0, 0x0, 0x0)
	/root/build/git-lfs-1.1.1-pre-push-tracing/obj-x86_64-linux-gnu/src/github.com/github/git-lfs/localstorage/localstorage.go:48 +0x81
github.com/github/git-lfs/lfs.LocalMediaPath(0x8afd30, 0x28, 0x0, 0x0, 0x0, 0x0)
	/root/build/git-lfs-1.1.1-pre-push-tracing/obj-x86_64-linux-gnu/src/github.com/github/git-lfs/lfs/lfs.go:56 +0x62
github.com/github/git-lfs/lfs.TestExistingUpload(0xc20804a900)
	/root/build/git-lfs-1.1.1-pre-push-tracing/obj-x86_64-linux-gnu/src/github.com/github/git-lfs/lfs/upload_test.go:109 +0x3f1
testing.tRunner(0xc20804a900, 0xc5fc28)
	/usr/lib/go/src/pkg/testing/testing.go:422 +0x8b
created by testing.RunTests
	/usr/lib/go/src/pkg/testing/testing.go:504 +0x8db

goroutine 16 [chan receive]:
testing.RunTests(0x905780, 0xc5f460, 0x61, 0x61, 0xc208000901)
	/usr/lib/go/src/pkg/testing/testing.go:505 +0x923
testing.Main(0x905780, 0xc5f460, 0x61, 0x61, 0xc55040, 0x1, 0x1, 0xc77b20, 0x0, 0x0)
	/usr/lib/go/src/pkg/testing/testing.go:435 +0x84
main.main()
	github.com/github/git-lfs/lfs/_test/_testmain.go:243 +0x9c

goroutine 19 [finalizer wait]:
runtime.park(0x4141c0, 0xc726d0, 0xc5e289)
	/usr/lib/go/src/pkg/runtime/proc.c:1369 +0x89
runtime.parkunlock(0xc726d0, 0xc5e289)
	/usr/lib/go/src/pkg/runtime/proc.c:1385 +0x3b
runfinq()
	/usr/lib/go/src/pkg/runtime/mgc0.c:2644 +0xcf
runtime.goexit()
	/usr/lib/go/src/pkg/runtime/proc.c:1445

goroutine 23 [chan receive]:
github.com/github/git-lfs/lfs.(*Batcher).acceptInput(0xc20800e520)
	/root/build/git-lfs-1.1.1-pre-push-tracing/obj-x86_64-linux-gnu/src/github.com/github/git-lfs/lfs/batcher.go:64 +0xbc
created by github.com/github/git-lfs/lfs.NewBatcher
	/root/build/git-lfs-1.1.1-pre-push-tracing/obj-x86_64-linux-gnu/src/github.com/github/git-lfs/lfs/batcher.go:26 +0xa3

goroutine 24 [chan receive]:
github.com/github/git-lfs/lfs.(*Batcher).acceptInput(0xc20800e5c0)
	/root/build/git-lfs-1.1.1-pre-push-tracing/obj-x86_64-linux-gnu/src/github.com/github/git-lfs/lfs/batcher.go:64 +0xbc
created by github.com/github/git-lfs/lfs.NewBatcher
	/root/build/git-lfs-1.1.1-pre-push-tracing/obj-x86_64-linux-gnu/src/github.com/github/git-lfs/lfs/batcher.go:26 +0xa3

goroutine 25 [chan send]:
github.com/github/git-lfs/lfs.(*Batcher).acceptInput(0xc20800e5e0)
	/root/build/git-lfs-1.1.1-pre-push-tracing/obj-x86_64-linux-gnu/src/github.com/github/git-lfs/lfs/batcher.go:72 +0x13b
created by github.com/github/git-lfs/lfs.NewBatcher
	/root/build/git-lfs-1.1.1-pre-push-tracing/obj-x86_64-linux-gnu/src/github.com/github/git-lfs/lfs/batcher.go:26 +0xa3

goroutine 27 [chan send]:
github.com/github/git-lfs/lfs.(*Batcher).acceptInput(0xc20800e660)
	/root/build/git-lfs-1.1.1-pre-push-tracing/obj-x86_64-linux-gnu/src/github.com/github/git-lfs/lfs/batcher.go:72 +0x13b
created by github.com/github/git-lfs/lfs.NewBatcher
	/root/build/git-lfs-1.1.1-pre-push-tracing/obj-x86_64-linux-gnu/src/github.com/github/git-lfs/lfs/batcher.go:26 +0xa3

goroutine 28 [chan send]:
github.com/github/git-lfs/lfs.(*Batcher).acceptInput(0xc20800e720)
	/root/build/git-lfs-1.1.1-pre-push-tracing/obj-x86_64-linux-gnu/src/github.com/github/git-lfs/lfs/batcher.go:72 +0x13b
created by github.com/github/git-lfs/lfs.NewBatcher
	/root/build/git-lfs-1.1.1-pre-push-tracing/obj-x86_64-linux-gnu/src/github.com/github/git-lfs/lfs/batcher.go:26 +0xa3

goroutine 29 [chan send]:
github.com/github/git-lfs/lfs.(*Batcher).acceptInput(0xc20800e760)
	/root/build/git-lfs-1.1.1-pre-push-tracing/obj-x86_64-linux-gnu/src/github.com/github/git-lfs/lfs/batcher.go:72 +0x13b
created by github.com/github/git-lfs/lfs.NewBatcher
	/root/build/git-lfs-1.1.1-pre-push-tracing/obj-x86_64-linux-gnu/src/github.com/github/git-lfs/lfs/batcher.go:26 +0xa3

goroutine 175 [runnable]:
net/http.(*Server).Serve(0xc2084541e0, 0x7f22f7f22400, 0xc208451410, 0xf, 0xc20800ff00)
	/usr/lib/go/src/pkg/net/http/server.go:1694
created by net/http/httptest.(*Server).Start
	/usr/lib/go/src/pkg/net/http/httptest/server.go:109 +0x2a3
FAIL	github.com/github/git-lfs/lfs	0.047s
?   	github.com/github/git-lfs/localstorage	[no test files]
?   	github.com/github/git-lfs/script	[no test files]
dh_auto_test: go test -v github.com/github/git-lfs github.com/github/git-lfs/commands github.com/github/git-lfs/docs/man github.com/github/git-lfs/git github.com/github/git-lfs/lfs github.com/github/git-lfs/localstorage github.com/github/git-lfs/script returned exit code 1
debian/rules:44: recipe for target 'override_dh_auto_test' failed
make[1]: *** [override_dh_auto_test] Error 1
make[1]: Leaving directory '/root/build/git-lfs-1.1.1-pre-push-tracing'
debian/rules:18: recipe for target 'build' failed
make: *** [build] Error 2
dpkg-buildpackage: error: debian/rules build gave error exit status 2
debuild: fatal error at line 1376:
dpkg-buildpackage -rfakeroot -D -us -uc failed
```

**Comment by tpdownes** [comment]
Oh, hey, you have packagecloud. Didn't see that because I went through the [releases page](https://github.com/github/git-lfs/releases/tag/v1.1.1). Can you add a link to the [packagecloud instructions](https://packagecloud.io/github/git-lfs) in the future release notes?

Nevertheless, I'm certain that the maintainers of our repository prefer to build things on their own - in part because we provide software to distributed users (we're part of a scientific collaboration) - and telling them to enable 1 repo we manage is easier than saying "enable this repo plus a whole bunch of other 3rd party repos which may/not enforce compatibility."

**Comment by tpdownes** [comment]
With 1.1.2 I now obtain this error 

```
=== RUN TestGitAndRootDirs
--- FAIL: TestGitAndRootDirs (0.00 seconds)
	git_test.go:278: Failed to call git rev-parse --git-dir --show-toplevel: "fatal: Not a git repository (or any of the parent directories): .git\n"
FAIL
FAIL	github.com/github/git-lfs/git	0.314s
```

Pleasantly, the previous failure appears to be resolved

```
=== RUN TestExistingUpload
--- PASS: TestExistingUpload (0.00 seconds)
```

**Comment by technoweenie** [comment]
Ah, that probably requires being run from inside a git repository.

**Comment by tpdownes** [comment]
Is that to say that you'll ensure that the build tests create a repo prior to running the test?

I note the presence of repos in the 1.1.2 release. Thanks!

**Comment by technoweenie** [comment]
The test is basically just running `git rev-parse --git-dir --show-toplevel` from the local directory. I made the assumption that the Golang package tests would always be run from a clone of this repository. There are some other assumptions too, such as the Go program should live in `$GOPATH/src/github.com/github/git-lfs` to compile and run properly.

I don't really plan to change this, though. This assumption makes sense for developers working on Git LFS. I can probably add an exception for that one test though.

**Comment by tpdownes** [comment]
Debian packages build from source tarballs, not from git repos. You should expect that to be an unsafe assumption.

**Comment by technoweenie** [comment]
Fixed in #1107. Thanks for the report.
